### ISSUE TITLE
Couchdb error - Purge checkpoint '_local/purge-mrview-' not updated in 59 seconds

### ISSUE THREAD

**Issue by aalegriadg** [apache/couchdb#4181]
When I make a [purge request](https://docs.couchdb.org/en/stable/api/database/misc.html) of 1000 documents, I get the following error logs:

```


[error] 2022-09-14T10:17:18.414447Z couchdb@127.0.0.1 <0.6046.12> -------- Purge checkpoint '_local/purge-mrview-12e57817980f2e08f39604bed34e44ad' not updated in 59 seconds
                    in database <<"shards/00000000-7fffffff/db.1659444354">>
[error] 2022-09-14T10:17:18.419849Z couchdb@127.0.0.1 <0.6046.12> -------- Purge checkpoint '_local/purge-mrview-279fd3dc8974ceb702c8b67421a6c856' not updated in 59 seconds
                    in database <<"shards/00000000-7fffffff/db.1659444354">>
[error] 2022-09-14T10:17:18.439182Z couchdb@127.0.0.1 <0.6046.12> -------- Purge checkpoint '_local/purge-mrview-2e64b0bb954e88d783d3d85f54ab79c4' not updated in 59 seconds
                    in database <<"shards/00000000-7fffffff/db.1659444354">>
[error] 2022-09-14T10:17:18.442420Z couchdb@127.0.0.1 <0.6046.12> -------- Purge checkpoint '_local/purge-mrview-33758fa40d7894fc9b86215711c23ca6' not updated in 59 seconds
                    in database <<"shards/00000000-7fffffff/db.1659444354">>
[error] 2022-09-14T10:17:18.457073Z couchdb@127.0.0.1 <0.6046.12> -------- Purge checkpoint '_local/purge-mrview-33d3a809ad09581167acfdae005e8989' not updated in 59 seconds
                    in database <<"shards/00000000-7fffffff/db.1659444354">>
[error] 2022-09-14T10:17:18.459602Z couchdb@127.0.0.1 <0.6046.12> -------- Purge checkpoint '_local/purge-mrview-78976a811e4bfad9f81151dd21139775' not updated in 59 seconds
                    in database <<"shards/00000000-7fffffff/db.1659444354">>
[error] 2022-09-14T10:17:18.462753Z couchdb@127.0.0.1 <0.6046.12> -------- Purge checkpoint '_local/purge-mrview-7da2929059e4620d3b669fad56a31dbd' not updated in 59 seconds
                    in database <<"shards/00000000-7fffffff/db.1659444354">>
[error] 2022-09-14T10:17:18.465246Z couchdb@127.0.0.1 <0.6046.12> -------- Purge checkpoint '_local/purge-mrview-85532e923a6bdaa9e16645c6c7e34fd0' not updated in 59 seconds
                    in database <<"shards/00000000-7fffffff/db1659444354">>

```
Purging is successful but the logs generate mistrust.

These errors only appear when I make a purge request for a large number of documents. If, for example, the request only includes the ids and revs of 100 documents, the error does not appear.

Why do these errors occur? Is this a problem? How can I avoid it?

**Comment by nickva** [comment]
See some documentation about https://docs.couchdb.org/en/stable/cluster/purging.html#config-settings

Wonder if your  `index_lag_warn_seconds = 86400` setting was altered from the default?

**Comment by aalegriadg** [comment]
No, the configuration is the default one: `index_lag_warn_seconds = 86400`

However, unlike described in the documentation, the seconds in my logs are not 86400. In the example of the description it appears 59 seconds but testing I have seen other values like: 1335, 1237, 126. 9 seconds ...

**Comment by nickva** [comment]
@aalegriadg You're right, thanks for checking. It looks like it's a bug in the logging bit. It logs how much over the threshold it is, and not the threshold limit itself as the message implied:

https://github.com/apache/couchdb/blob/96214780e891b173f1f8049d50206fcb097c26a5/src/couch/src/couch_db.erl#L525-L529

It does look like a warning only, and indicates that the views have not been updated (yet?) after the documents have been purged from the main db. Try to query the views (which forces an update) or perhaps adjust the auto-indexer, [ken](https://docs.couchdb.org/en/stable/config/indexbuilds.html) settings.

What I suspect might be happening is that we don't have have a decent auto-indexer trigger based on purged only. Only on document updates.

**Comment by aalegriadg** [comment]
Thank you for your answer, it is very useful for my team.

> You're right, thanks for checking. It looks like it's a bug in the logging bit. It logs how much over the threshold it is, and not the threshold limit itself as the message implied:

As it is a bug, this log should only appear if the purge takes more than 1 day (86400 seconds), which is not going to happen. So, do I stop worrying about these warnings?

> It does look like a warning only, and indicates that the views have not been updated (yet?) after the documents have been purged from the main db. Try to query the views (which forces an update) or perhaps adjust the auto-indexer, ken settings.

As it is a partitioned database, the reindexing only affects the documents belonging to the queried partition right? i.e. if I purge partition1 the query time of partition2 should not be affected. I ask this because in my project we work with temporary replication windows: Documents are stored by daily partitions and when they are considered old enough they are not replicated and furthermore, they are purged. The query is for the partition with recent documents (those of the current day), not the purged ones. By the way I also opened [this issue ](https://github.com/apache/couchdb/issues/4180)about purging.
